polyfactory/factories/base.py

```diff
@@ -321,13 +327,18 @@ def _handle_factory_field(cls, field_value: Any, field_build_parameters: Any | N
     def _get_or_create_factory(
         cls,
         model: type,
-    ) -> type["BaseFactory"]:
+    ) -> type[BaseFactory]:
         """Get a factory from registered factories or generate a factory dynamically.
 
         :param model: A model type.
         :returns: A Factory sub-class.
 
         """
+        if cls.__base_factory_overrides__ and (
+            factory := cls.__base_factory_overrides__.get(model, cls.__base_factory_overrides__.get(type(model)))
```

Comment:
`BaseModel` is the base type of the model, I guess you mean `pydantic.main.ModelMetaclass`? And the idea is that a user could say `__base_factory_overrides__ = {ModelMetaclass: FooFactory}` to mean "use `FooFactory` to handle all pydantic models"?

Reply:
No, i mean - consider the relation between a pydantic model and `BaseModel`. 


```python
class X(BaseModel):
   pass
```

The `type` of X is `BaseModel`.

Reply:
No it's not, that's what confused me:
```python
In [1]: from pydantic import BaseModel

In [2]: class X(BaseModel):
   ...:    pass
   ...:

In [3]: type(X)
Out[3]: pydantic.main.ModelMetaclass
```

Reply:
hmm, you're right. Well, i would be happy for a followup PR if you wanna submit one - the solution I added is not very robust as implemented, and if you wanna make it more robust and traverse the MRO chain it would be great.